plugins/auth-backend/src/service/router.ts

```diff
@@ -249,3 +249,22 @@ export function createOriginFilter(
     return allowedOriginPatterns.some(pattern => pattern.match(origin));
   };
 }
+
+/** @internal */
+export function getDefaultBackstageTokenExpiryTime(config: Config) {
+  const processingIntervalKey = 'auth.backstageTokenExpiration';
+
+  if (!config.has(processingIntervalKey)) {
+    return BACKSTAGE_SESSION_EXPIRATION;
+  }
+
+  const duration = readDurationFromConfig(config, {
+    key: processingIntervalKey,
+  });
+  const seconds = Math.max(
+    600,
+    Math.round(durationToMilliseconds(duration) / 1000),
+  );
```

Comment:
Could we do a max of 24h here as well, or something similarly sensible? There is no revocation of these tokens and their scope is wide, so I think we need to have an upper limit in place as well.